src/include/kdb.h.in

```diff
@@ -134,80 +125,61 @@ int kdbSet (KDB *handle, KeySet *returned,
 
-int keyNeedSync (const Key *key);
+/* Relations */
+int elektraKeyCompareName (const ElektraKey * k1, const ElektraKey * k2);
 
-int keyIsBelow (const Key *key, const Key *check);
-int keyIsBelowOrSame (const Key *key, const Key *check);
-int keyIsDirectlyBelow (const Key *key, const Key *check);
-
-int keyIsBinary (const Key *key);
-int keyIsString (const Key *key);
+int elektraKeyIsBelow (const ElektraKey * key, const ElektraKey * check);
+int elektraKeyIsBelowOrSame (const ElektraKey * key, const ElektraKey * check);
+int elektraKeyIsDirectlyBelow (const ElektraKey * key, const ElektraKey * check);
 
 /* Name Manipulation Methods */
-const char *keyName (const Key *key);
-ssize_t keyGetNameSize (const Key *key);
-ssize_t keyGetName (const Key *key, char *returnedName, size_t maxSize);
-
-ssize_t keySetName (Key *key, const char *newname);
-ssize_t keyAddName (Key *key, const char *addName);
+const char * elektraKeyName (const ElektraKey * key);
+ssize_t elektraKeyNameSize (const ElektraKey * key);
 
-const void *keyUnescapedName (const Key *key);
-ssize_t keyGetUnescapedNameSize (const Key *key);
+ssize_t elektraKeySetName (ElektraKey * key, const char * newname);
+ssize_t elektraKeyAddName (ElektraKey * key, const char * addName);
 
-const char *keyBaseName (const Key *key);
-ssize_t keyGetBaseNameSize (const Key *key);
-ssize_t keyGetBaseName (const Key *key, char *returned, size_t maxSize);
+const char * elektraKeyEscapedName (const ElektraKey * key);
 
-ssize_t keySetBaseName (Key *key,const char *baseName);
-ssize_t keyAddBaseName (Key *key,const char *baseName);
+const char * elektraKeyBaseName (const ElektraKey * key);
+ssize_t elektraKeyBaseNameSize (const ElektraKey * key);
 
-elektraNamespace keyGetNamespace (Key const* key);
-ssize_t keySetNamespace (Key * key, elektraNamespace ns);
+ssize_t elektraKeySetBaseName (ElektraKey * key, const char * baseName);
+ssize_t elektraKeyAddBaseName (ElektraKey * key, const char * baseName);
 
-const void *keyValue (const Key *key);
-ssize_t keyGetValueSize (const Key *key);
+elektraNamespace elektraKeyNamespace (ElektraKey const * key);
+ssize_t elektraKeySetNamespace (ElektraKey * key, elektraNamespace ns);
 
-const char *keyString (const Key *key);
-ssize_t keyGetString (const Key *key, char *returnedString, size_t maxSize);
-ssize_t keySetString (Key *key, const char *newString);
+/* Value Manipulation Methods */
+const void * elektraKeyValue (const ElektraKey * key);
+ssize_t elektraKeyValueSize (const ElektraKey * key);
 
-ssize_t keyGetBinary (const Key *key, void *returnedBinary, size_t maxSize);
-ssize_t keySetBinary (Key *key, const void *newBinary, size_t dataSize);
+ssize_t elektraKeySetValue (ElektraKey * key, const void * value, size_t valueSize);
 
-int keyLock (Key * key, elektraLockFlags what);
-int keyIsLocked (const Key * key, elektraLockFlags what);
+/* Locking */
+int elektraKeyLock (ElektraKey * key, elektraLockFlags what);
```

Comment:
With the new automatic locking on keynames, it might make sense to have a separate function for that to highlight that locking of names works differently. But 3 separate functions also seems excessive.